src/datasources/network/network.module.ts

```diff
@@ -1,24 +1,55 @@
 import { Global, Module } from '@nestjs/common';
-import axios, { Axios } from 'axios';
-import * as http from 'http';
-import * as https from 'https';
 import { IConfigurationService } from '@/config/configuration.service.interface';
-import { AxiosNetworkService } from '@/datasources/network/axios.network.service';
+import { FetchNetworkService } from '@/datasources/network/fetch.network.service';
 import { NetworkService } from '@/datasources/network/network.service.interface';
 
+export type FetchClient = <T>(
+  url: string,
+  options: RequestInit,
+) => Promise<{ data: T }>;
+
 /**
- * Use this factory to add any default parameter to the
- * {@link Axios} instance
+ * Use this factory to create a {@link FetchClient} instance
+ * that can be used to make HTTP requests.
  */
-function axiosFactory(configurationService: IConfigurationService): Axios {
+function fetchClientFactory(
+  configurationService: IConfigurationService,
+): FetchClient {
   const requestTimeout = configurationService.getOrThrow<number>(
     'httpClient.requestTimeout',
   );
-  return axios.create({
-    timeout: requestTimeout,
-    httpAgent: new http.Agent({ keepAlive: true }),
-    httpsAgent: new https.Agent({ keepAlive: true }),
-  });
+
+  return async <T>(url: string, options: RequestInit): Promise<{ data: T }> => {
+    // If general error - NetworkOtherError
+    const response = await fetch(url, {
+      ...options,
+      signal: AbortSignal.timeout(requestTimeout),
+      keepalive: true,
+    });
```

Comment:
If an error is thrown in the `fetch` then this error will propagate upwards instead of being mapped to our `NetworkRequestError`

Reply:
This is now caught and thrown correctly as of 3c3a5f5053f1d8d0524d13cfbe9c2302153f6572.